drivers/sensor/bmm150/bmm150.c

```diff
@@ -483,28 +516,27 @@ static const struct sensor_driver_api bmm150_api_funcs = {
 static int bmm150_init_chip(const struct device *dev)
 {
 	struct bmm150_data *data = dev->data;
-	const struct bmm150_config *config = dev->config;
 	uint8_t chip_id;
 	struct bmm150_preset preset;
 
-	if (bmm150_set_power_mode(dev, BMM150_POWER_MODE_NORMAL, 0) < 0) {
-		LOG_ERR("failed to bring up device from normal mode");
-		return -EIO;
-	}
-
-	if (bmm150_set_power_mode(dev, BMM150_POWER_MODE_SUSPEND, 1) < 0) {
-		LOG_ERR("failed to bring up device in suspend mode");
-		return -EIO;
+	/* Soft reset chip */
+	if (bmm150_reg_update_byte(dev, BMM150_REG_POWER, BMM150_MASK_SOFT_RESET,
+				   BMM150_SOFT_RESET) < 0) {
+		LOG_ERR("failed reset chip");
+		goto err_poweroff;
 	}
+	k_busy_wait(1000);
```

Comment:
Please document why this particular wait duration was chosen. Can you sleep instead of busy-wait?

Reply:
> Please document why this particular wait duration was chosen. Can you sleep instead of busy-wait?

wait 1ms is comes from here:
https://github.com/boschsensortec/BMM150-Sensor-API/blob/a20641f216057f0c54de115fe81b57368e119c01/bmm150.c#L677

Reply:
> Please document why this particular wait duration was chosen. Can you sleep instead of busy-wait?

From suspend mode to sleep mode, it need sleep for 3ms, it comes here:
https://github.com/boschsensortec/BMM150-Sensor-API/blob/a20641f216057f0c54de115fe81b57368e119c01/bmm150.c#L1287